lib/src/json_schema/json_schema.dart

```diff
@@ -2052,27 +2056,27 @@ class JsonSchema {
     }
   }
 
-  /// Mixin another JsonSchema into this one. All references must be resolved before calling.
-  mixinForRef(JsonSchema ref) {
-    this._schemaMap.remove(r'$ref');
-    this._ref = null;
-    // The specification might be ambiguous on how to merge references into the current node. This is our best guess.
-    this._schemaMap.deepMerge(ref._schemaMap);
-
-    this._validateAndSetAllProperties();
-  }
-
   /// Find the furthest away parent [JsonSchema] the that is a recursive anchor
   /// or null of there is no recursiveAnchor found.
   JsonSchema furthestRecursiveAnchorParent() {
     JsonSchema lastFound = this.recursiveAnchor ? this : null;
-    var possibleAnchor = this._parent;
+    var possibleAnchor = this.dynamicParent;
     while (possibleAnchor != null) {
       if (possibleAnchor.recursiveAnchor) {
         lastFound = possibleAnchor;
       }
-      possibleAnchor = possibleAnchor._parent;
+      possibleAnchor = possibleAnchor.dynamicParent;
     }
     return lastFound;
   }
+
+  void pushDynamicParent(JsonSchema value) {
+    if (value != this) {
+      _dynamicParents.add(value);
+    }
+  }
+
+  void popDynamicParent() {
+    _dynamicParents.removeLast();
+  }
```

Comment:
I don't really like modifying the JsonSchema objects at validation time, but it works for these tests....

Reply:
I've been contemplating if there was an easy way to do that without refactoring a bunch of code....

Reply:
Yes, there is.